### PR TITLE
fix: fixed query param not going for delete

### PR DESCRIPTION
**Pull request by manisha1997**
# Fixes #

Fix done: HTTP client will now send params and data both

### Checklist
- [x] I acknowledge that all my contributions will be made under the project's license
- [x] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [x] I have read the [Contribution Guidelines](https://github.com/twilio/twilio-ruby/blob/main/CONTRIBUTING.md) and my PR follows them
- [x] I have titled the PR appropriately
- [x] I have updated my branch with the main branch
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary documentation about the functionality in the appropriate .md file
- [ ] I have added inline documentation to the code I modified

If you have questions, please file a [support ticket](https://twilio.com/help/contact), or create a GitHub Issue in this repository.

### CODE REVIEW

**Comment by manisha1997** [comment]
@shrutiburman , @tiwarishubham635 , @AsabuHere and @sbansla 
I tried writing test case for this like:
`  it 'should contain params in Connection object' do
    expect(Faraday).to receive(:new).and_return(Faraday::Connection.new)
    allow_any_instance_of(Faraday::Connection).to receive(:send).and_return(double('response', status: 301, body: {}, headers: { something: '1' }))
    params = Twilio::Values.of({"PhoneNumber" => "+18001234567"})
    @client.request('host', 'port', 'DELETE', 'url', params, nil, {'abc':'xyz'}, %w[a b])
    expect(@client.connection.params).to eq({'abc' => 'xyz'})
  end`

This code is working well in manual testing but the test case is failing and the params is becoming nil. This is not the case with manual testing but in unit testing. If someone has some idea about Faraday, can you guys help me out?

**Review by tiwarishubham635 — APPROVED** [review]
LGTM!

File: lib/twilio-ruby/http/http_client.rb [hunk 1]
```diff
@@ -6,7 +6,7 @@ module Twilio
   module HTTP
     class Client
       attr_accessor :adapter
-      attr_reader :timeout, :last_response, :last_request
+      attr_reader :timeout, :last_response, :last_request, :connection
```
Comment (shrutiburman) [inline lib/twilio-ruby/http/http_client.rb:9]: Why do we need connection here, when we are creating one in line 30, while making request ?

Comment (manisha1997) [reply lib/twilio-ruby/http/http_client.rb:9]: The connection variable was always there, I made the variable readable to use it in unit testing

**Pull request merged**